columnflow/tasks/framework/mixins.py

```diff
@@ -2157,8 +1936,173 @@ def processes_repr(self):
+
+            params["processes"] = tuple(processes)
+            params["process_insts"] = tuple(
+                tuple(config_inst.get_process(p) for p in params["processes"][i])
+                for i, config_inst in enumerate(config_insts)
+            )
+
+        # resolve datasets
+        if "datasets" in params:
+            if params["datasets"]:
+                datasets = list(params["datasets"])
+                if len(datasets) == 1:
+                    # resolve to number of configs
+                    datasets = list(datasets * len(config_insts))
+                elif len(datasets) != n_configs:
+                    raise ValueError(
+                        f"number of datasets ({len(datasets)}) does not match number of configs ({n_configs})",
+                    )
+
+                for i, _datasets in enumerate(datasets):
+                    config_inst = config_insts[i]
+                    datasets[i] = tuple(cls.find_config_objects(
+                        _datasets,
+                        config_inst,
+                        od.Dataset,
+                        config_inst.x("dataset_groups", {}),
+                    ))
+
+            elif "processes" in params:
+                # pick all datasets that contain any of the requested (sub) processes
+                datasets = list()
+                for i, _processes in enumerate(processes):
+
+                    sub_process_insts = sum((
+                        [proc for proc, _, _ in process_inst.walk_processes(include_self=True)]
+                        for process_inst in map(config_insts[i].get_process, _processes)
+                    ), [])
+                    datasets.append(tuple(
+                        dataset_inst.name
+                        for dataset_inst in config_insts[i].datasets
+                        if any(map(dataset_inst.has_process, sub_process_insts))
+                    ))
+
+            # complain when no datasets were found
+            if not datasets and not cls.allow_empty_datasets:
+                raise ValueError(f"no datasets found matching {params['datasets']}")
+
+            params["datasets"] = tuple(datasets)
+            params["dataset_insts"] = tuple(
+                tuple(config_inst.get_dataset(d) for d in params["datasets"][i])
+                for i, config_inst in enumerate(config_insts)
+            )
+            params["configs_vs_dataset_insts"] = dict(zip([c.name for c in config_insts], params["dataset_insts"]))
+
+        return params
+
+    @classmethod
+    def get_known_shifts(cls, config_inst, params):
+        shifts, upstream_shifts = super().get_known_shifts(config_inst, params)
+
+        # add shifts of all datasets to upstream ones
+        for dataset_inst in params.get("configs_vs_dataset_insts")[config_inst.name] or []:
+            if dataset_inst.is_mc:
+                upstream_shifts |= set(dataset_inst.info.keys())
```

Comment:
Do we really need this distinction for mc datasets? The code itself should also work for observed data, right? I'm asking because I think we should keep the list of presumptions as short as possible, and you never know - maybe there will some day be a varied dataset for observed data, which might correspond to some weird technique future analyzers might need at some point. Though now that I think about it, this might also break all of the code in other places, so it might be necessary after all

Reply:
This is just copy-pasted from the `DatasetsProcessesMixin`, so I'd prefer to not touch this in this PR